src/Microsoft.OData.Core/UrlValidation/ODataUrlValidationExtensions.cs

```diff
@@ -0,0 +1,61 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataUrlValidationExtension.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using Microsoft.OData.Edm;
+using Microsoft.OData.UriParser.Validation.ValidationEngine;
+
+namespace Microsoft.OData.UriParser.Validation
+{
+    /// <summary>
+    /// Extension methods to add validation methods to an ODataUri
+    /// </summary>
+    public static class ODataUrlValidationExtensions
+    {
+        /// <summary>
+        /// Validate the OData Url for a given model using a specified set of rules.
+        /// </summary>
+        /// <param name="uri">The <see cref="Uri"/> to validate.</param>
+        /// <param name="model">The model to validate the OData Uri against.</param>
+        /// <param name="rules">The set of rules to use in validating the OData Uri.</param>
+        /// <param name="validationMessages">The collection of validation messages found during validation.</param>
+        /// <returns>True if validation messages are discovered during validation, otherwise false.</returns>
+        public static bool ValidateODataUrl(this Uri uri, IEdmModel model, ODataUrlValidationRuleSet rules, out IEnumerable<ODataUrlValidationMessage> validationMessages)
+        {
+            try
+            {
+                ODataUriParser parser = new ODataUriParser(model, uri);
+                ODataUri odataUri = parser.ParseUri();
+                return odataUri.Validate(model, rules, out validationMessages);
+            }
+
+            catch (Exception e)
+            {
+                validationMessages = new ODataUrlValidationMessage[]
+                {
+                    new ODataUrlValidationMessage(ODataUrlValidationMessageCodes.UnableToParseUri, e.Message, Severity.Error)
+                };
+
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Validate the ODataUri for a given model using a specified set of rules.
+        /// </summary>
+        /// <param name="odataUri">The <see cref="ODataUri"/> to validate.</param>
+        /// <param name="model">The model to validate the OData Uri against.</param>
+        /// <param name="rules">The set of rules to use in validating the OData Uri.</param>
+        /// <param name="validationMessages">The collection of validation messages found during validation.</param>
+        /// <returns>True if validation messages are discovered during validation, otherwise false.</returns>
+        public static bool Validate(this ODataUri odataUri, IEdmModel model, ODataUrlValidationRuleSet rules, out IEnumerable<ODataUrlValidationMessage> validationMessages)

```

Comment:
I'd like to have "Validate" overload without ODataUrlValidationRuleSet, That's to use the built-in default RuleSet.

Reply:
Especially since our default ruleset only has two types of rules today, and adding rules in the future would potentially make urls that current validate become invalid, I'd rather keep this explicit. It's easy enough to pass in ODataValidationRuleSet.AllRules to explicitly validate according to all rules.

---
In reply to: [434870502](https://github.com/OData/odata.net/pull/1769#discussion_r434870502) [](ancestors = 434870502)